### PR TITLE
[BUG] Orbit camera instead of rotating objects in 3D view

### PR DESCRIPTION
**Pull request by alexrockhill**
Sorry this took so long to have a decent enough start to commit @LorenzE! I think I have the lay of the land and can work on a solution. I'm trying to implement it like here https://doc.qt.io/qt-5/qt3d-simple-cpp-example.html (I tried first with `QCamera::panAboutViewCenter` but I guess you can't animate functions only properties?). It's still crashing but I was going to work on it for another hour or so and then maybe ask for help if it's still breaking.

### CODE REVIEW

**Comment by alexrockhill** [comment]
Hmmm it's crashing without an error so I'll have to figure out the debugger...

**Comment by alexrockhill** [comment]
Ok, ready for review!

**Comment by LorenzE** [comment]
I would suggest making the OrbitalCameraController’s property to take in the angle to rotate in degrees instead of a plain int. This way the rotate functions are a bit more descriptive and more functional. You would also be able to remove the member variable `m_rotating`.

**Comment by alexrockhill** [comment]
Ok so I checked and you get a warning if the `QProperty` doesn't have a READ function and you get an error if you try not to pass a value to `QPropertyAnimation` so I think this is the best that can be done. Alternatively, we could return `null` for reading in the property `rotating` but that seems more confusing than the way it is now.

File: libraries/disp3D/engine/view/orbitalcameracontroller.cpp [hunk 1]
```diff
@@ -138,3 +140,17 @@ void OrbitalCameraController::initController()
...
+
+void OrbitalCameraController::setRotating(int count)
+{
+    Qt3DRender::QCamera *pCamera = this->camera();
+
+    m_rotating = count;
+    QQuaternion quat = QQuaternion::QQuaternion::fromEulerAngles(0,0,m_fAutoRotationSpeed);
+    pCamera->rotateAboutViewCenter(quat);
+}
+
```
Comment (LorenzE) [inline libraries/disp3D/engine/view/orbitalcameracontroller.cpp:152]: Please add spacer
//=============================================================================================================

File: libraries/disp3D/engine/view/orbitalcameracontroller.h [hunk 2]
```diff
@@ -107,6 +115,13 @@ class DISP3DSHARED_EXPORT OrbitalCameraController : public Qt3DExtras::QAbstract
      */
     void invertCameraRotation(bool newStatusFlag);
 
+    //=========================================================================================================
+    /**
+     * Sets the angle of the camera for rotating around
+     */
+    void setRotating(int count);     /* The counter for how long the rotation has been happening. */
+    int rotating() const;
```
Comment (LorenzE) [inline libraries/disp3D/engine/view/orbitalcameracontroller.h:123]: please document each function and their functions values (input and return)

File: libraries/disp3D/engine/view/orbitalcameracontroller.h [hunk 3]
```diff
@@ -136,6 +151,8 @@ class DISP3DSHARED_EXPORT OrbitalCameraController : public Qt3DExtras::QAbstract
 
     float m_rotationInversFactor = 1.0f;             /**< The factor used to invers the camera rotation. */
```
Comment (LorenzE) [inline libraries/disp3D/engine/view/orbitalcameracontroller.h:152]: Please mind indentation for start of docu /**<

Comment (LorenzE) [inline libraries/disp3D/engine/view/orbitalcameracontroller.h:152]: Although you did not add this line, please rename `m_rotationInversFactor` to `m_fRotationInverseFactor`. the `f` hereby indicates a float value

File: libraries/disp3D/engine/view/orbitalcameracontroller.h [hunk 4]
```diff
@@ -136,6 +151,8 @@ class DISP3DSHARED_EXPORT OrbitalCameraController : public Qt3DExtras::QAbstract
...
     const float m_fZoomInLimit = 0.04f;         /**< The minimum distance of the camera to the the view center. */
+    const float m_fAutoRotationSpeed = 1.0f;         /**< The speed that automatic rotation rotates */
+    int m_rotating;                      /**< How long the camera has been rotating with regards to the view center. */
```
Comment (LorenzE) [inline libraries/disp3D/engine/view/orbitalcameracontroller.h:155]: Please rename to m_iRotating

File: libraries/disp3D/engine/view/view3D.h [hunk 5]
```diff
@@ -122,9 +123,9 @@ class DISP3DSHARED_EXPORT View3D : public Qt3DExtras::Qt3DWindow
 
     //=========================================================================================================
     /**
-     * Starts or stops to rotate all loaded 3D models.
+     * Starts or stops to rotate camera around 3D models.
      */
-    void startStopModelRotation(bool checked);
+    void startStopCameraRotation(bool checked);
```
Comment (LorenzE) [inline libraries/disp3D/engine/view/view3D.h:128]: rename `checked` to `bChecked`

File: libraries/disp3D/engine/view/view3D.h [hunk 6]
```diff
@@ -51,6 +51,7 @@
 #include <QPointer>
 #include <QObjectPicker>
 #include <Qt3DCore>
+#include "orbitalcameracontroller.h"
```
Comment (LorenzE) [inline libraries/disp3D/engine/view/view3D.h:54]: move `#include "orbitalcameracontroller.h"` to line 44 right after `#include "../../disp3D_global.h"`

**Review by LorenzE — CHANGES_REQUESTED** [review]
Looks good to me. Just some minor comments

File: libraries/disp3D/engine/view/orbitalcameracontroller.h (hunk 2 above)
Comment (alexrockhill) [reply libraries/disp3D/engine/view/orbitalcameracontroller.h:123]: I forgot that one! I wasn't sure when I was trying things if they would work so I didn't document but I forgot to go back and fix it

**Comment by alexrockhill** [comment]
Ok should be good to go!

**Review by LorenzE — APPROVED** [review]
Nice work @alexrockhill

**Pull request merged**